File: old-python-src/debate_pkg/actors/critic.py
"""
Critic — rebuilt fresh each round, primed via system prompt with:
  - prior rephrased questions
  - active Answerer profile (notes seen >= threshold)
"""

from .base import Actor
from ..personas import (
    load_persona,
    render_prior_rephrased,
    render_answerer_profile,
)


class Critic(Actor):
    role = "critic"
    display_name = "Critic"

    def temperature(self):
        return self.ctx.config.critic_temp

    def render_system_prompt(self):
        template = load_persona(self.ctx.config.persona_name, self.role)
        return (
            template
            .replace("{prior_rephrased}", render_prior_rephrased(self.ctx.prior_rephrased))
            .replace("{answerer_profile}", render_answerer_profile(self.ctx.active_profile()))
        )

File: src/Debate.Core/Actors/Critic.cs
namespace Debate.Core.Actors;

/// <summary>
/// The Critic stress-tests the Answerer's position. Rebuilt fresh each round so
/// its system prompt is primed with the latest prior rephrased questions and the
/// active Answerer profile. It only ever sees the Judge's restatements.
/// </summary>
public sealed class Critic : Actor
{
    public Critic(DebateContext context) : base(context)
    {
    }

    public override DebateRole Role => DebateRole.Critic;
    public override string DisplayName => "Critic";

    protected override string RenderSystemPrompt()
    {
        var template = Context.Personas.Load(Context.Config.PersonaName, PersonaToken);
        return template
            .Replace("{prior_rephrased}", PersonaLibrary.RenderPriorRephrased(Context.PriorRephrased))
            .Replace("{answerer_profile}", PersonaLibrary.RenderAnswererProfile(Context.ActiveProfile()));
    }
}
